pkg/menu/menu.go

```diff
@@ -0,0 +1,73 @@
+package menu
+
+import (
+	"os"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+const menuWidth = 50
+
+// newParagraph will return a paragraph object with given initial text.
+func newParagraph(initText string, border bool, location int, wid int, ht int) *widgets.Paragraph {
+	p := widgets.NewParagraph()
+	p.Text = initText
+	p.Border = border
+	p.SetRect(0, location, wid, location+ht)
+	p.TextStyle.Fg = ui.ColorWhite
+	ui.Render(p)
+	return p
+}
+
+func processInput(input, warning *widgets.Paragraph, isValid func(string) (bool, string), uiEvents <-chan ui.Event) (string, error) {
+	// keep tracking all input from user
+	for {
+		e := <-uiEvents
+		if e.Type != ui.KeyboardEvent {
+			continue
+		}
+		switch e.ID {
+		// if user input control-c, exit the program.
+		case "<C-c>":
```

Comment:
I think C-d would be better, that is the usual character used in Unix meaning no more input.

Also, the convention for programs like this that take input is to return io.Eof, not just exit. The calling program might wish to do cleanup of its own.